database/ritadb.go

```diff
@@ -0,0 +1,185 @@
+package database
+
+import (
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/blang/semver"
+)
+
+type (
+	// RITADatabase provides methods for manipulating an
+	// entry in the RITADatabaseIndex
+	RITADatabase struct {
+		indexDoc            DBMetaInfo
+		metaDatabaseName    string
+		indexCollectionName string
```

Comment:
Bit of data duplication here. Chose this over holding a pointer to the RITADatabaseIndex object which created it.